thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java

```diff
@@ -217,73 +218,123 @@ public Response computeSeverity(@NotNull @QueryParam("collection") String collec
-    if (anomalyFunction == null) {
+    final List<Long> functionIdList = new ArrayList<>();
+    for (String functionId : ids.split(",")) {
+      AnomalyFunctionDTO anomalyFunction = anomalyFunctionDAO.findById(Long.valueOf(functionId));
+      if (anomalyFunction != null && anomalyFunction.getIsActive()) {
+        functionIdList.add(Long.valueOf(functionId));
+      } else {
+        // Check if the anomaly function is active
+        if (!anomalyFunction.getIsActive()) {
+          throw new IllegalArgumentException(String.format("Skipping deactivated function %s", functionId));
+        } else {
+          LOG.warn("Unable to load function id {}", functionId);
+        }
+      }
+    }
+    if (functionIdList.isEmpty()) {
       return Response.noContent().build();
     }
 
     // Check if the timestamps are available
     DateTime startTime = null;
     DateTime endTime = null;
     if (startTimeIso == null || startTimeIso.isEmpty()) {
-      throw new IllegalArgumentException(String.format("[functionId %s] Monitoring start time is not found", id));
+      LOG.error("[Backfill] Monitoring start time is not found");
+      throw new IllegalArgumentException(String.format("[Backfill] Monitoring start time is not found"));
     }
     if (endTimeIso == null || endTimeIso.isEmpty()) {
-      throw new IllegalArgumentException(String.format("[functionId %s] Monitoring end time is not found", id));
+      LOG.error("[Backfill] Monitoring end time is not found");
+      throw new IllegalArgumentException(String.format("[Backfill] Monitoring end time is not found"));
     }
 
     startTime = ISODateTimeFormat.dateTimeParser().parseDateTime(startTimeIso);
     endTime = ISODateTimeFormat.dateTimeParser().parseDateTime(endTimeIso);
 
     if (startTime.isAfter(endTime)) {
+      LOG.error("[Backfill] Monitoring start time is after monitoring end time");
       throw new IllegalArgumentException(String.format(
-          "[functionId %s] Monitoring start time is after monitoring end time", id));
+          "[Backfill] Monitoring start time is after monitoring end time"));
     }
     if (endTime.isAfterNow()) {
-      throw new IllegalArgumentException(String.format(
-          "[functionId %s] Monitor end time {} should not be in the future", id, endTime.toString()));
-    }
+      endTime = DateTime.now();
+      LOG.warn("[Backfill] End time is in the future. Force to now.");
+    }
+
+    final DateTime innerStartTime = startTime;
+    final DateTime innerEndTime = endTime;
+    new Thread(new Runnable() {
+      @Override
+      public void run() {
+        Map<Long, Integer> originalWindowSize = new HashMap<>();
+        Map<Long, TimeUnit> originalWindowUnit = new HashMap<>();
+        Map<Long, String> originalCron = new HashMap<>();
+        saveFunctionWindow(functionIdList, originalWindowSize, originalWindowUnit, originalCron);
+
+        // Update speed-up window and cron
+        if (speedup) {
```

Comment:
This is not generic for all functions, so it does not make sense to do this type of speedup for all. 

1. please add more inline comments to state why you choose these 'speedup' configurations and when it should be used.
2. when you do speedup this way for multiple functions, you have even stronger assumption here for your input, which should be explained in java doc.